lib/util/source-code.js

```diff
@@ -137,6 +140,24 @@ class SourceCode extends TokenStore {
          */
         this.ast = ast;
 
+        /**
+         * The parser services of this source code.
+         * @type {Object}
+         */
+        this.parserServices = parserServices || null;
+
+        /**
+         * The scope of this source code.
+         * @type {ScopeManager}
+         */
+        this.scopeManager = scopeManager || null;
```

Comment:
The first argument of `linter#verify` can be a `SourceCode` instance. In that case, parsing is skipped and the `SourceCode` instance is reused. So the `SourceCode` instance must have all result of parsing.